src/Cardano/Wallet/Api/Types/WalletAddressPoolGap.hs

```diff
@@ -0,0 +1,34 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module Cardano.Wallet.Api.Types.WalletAddressPoolGap
+    ( WalletAddressPoolGap
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.AddressDiscovery
+    ( AddressPoolGap, getAddressPoolGap, mkAddressPoolGap )
+import Data.Aeson
+    ( FromJSON, ToJSON, Value (Number), parseJSON, toJSON )
+import GHC.Generics
+    ( Generic )
+
+newtype WalletAddressPoolGap = WalletAddressPoolGap
+    { getWalletAddressPoolGap :: AddressPoolGap }
+    deriving stock (Generic, Show)
+    deriving newtype (Bounded, Enum, Eq, Ord)
```

Comment:
Not sure we need the `Bounded` and `Enum` here :thinking:

Reply:
They are used in `Cardano.Wallet.Api.Arbitraries`:
```hs
instance Arbitrary WalletAddressPoolGap where                                                                                                                                         
    arbitrary = arbitraryBoundedEnum    
```
This particular `Arbitrary` instance is used in JSON round-trip testing.